appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/EJBContainerTransactionManager.java

```diff
@@ -331,42 +320,45 @@ private void startNewTx(Transaction prevTx, EjbInvocation inv) throws Exception
      * Use caller transaction to execute a bean method
      */
     protected void useClientTx(Transaction prevTx, EjbInvocation inv) {
-        Transaction clientTx;
-        int status=-1;
-        int prevStatus=-1;
+        LOG.finest(() -> String.format("useClientTx(prevTx=%s, inv=%s)", prevTx, inv));
+        final Transaction clientTx;
+        final int status;
+        final int prevStatus;
         try {
             // Note: inv.clientTx will not be set at this point.
             clientTx = transactionManager.getTransaction();
             status = clientTx.getStatus();  // clientTx cant be null
-            if ( prevTx != null ) {
+            if (prevTx == null) {
+                prevStatus = -1;
+            } else {
                 prevStatus = prevTx.getStatus();
             }
         } catch (Exception ex) {
+            final TransactionRolledbackLocalException toThrow = new TransactionRolledbackLocalException(ex.getMessage(), ex);
             try {
                 transactionManager.setRollbackOnly();
-            } catch ( Exception e ) {
-		//FIXME: Use LogStrings.properties
-		_logger.log(Level.FINEST, "", e);
-	    }
-            throw new TransactionRolledbackLocalException("", ex);
+            } catch (Exception e) {
```

Comment:
Logging of exception as finest (so practically ignoring it) replaced with the usage of suppressed. It will be visible in a stacktrace.